taskcluster/async/asyncclient.py

```diff
@@ -0,0 +1,100 @@
+"""This module is used to interact with taskcluster rest apis"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import asyncio
+import aiohttp
+
+from contextlib import contextmanager
+
+import taskcluster.exceptions as exceptions
+import taskcluster.baseclient as baseclient
+import taskcluster.utils as utils
+import taskcluster.asyncutils as asyncutils
+
+log = logging.getLogger(__name__)
+
+
+class AsyncClient(baseclient.BaseClient):
+    """Async class for API Client Classes. Each individual Client class
+    needs to set up its own methods for REST endpoints and Topic Exchange
+    routing key patterns.
+    """
+
+    @contextmanager
+    def contextSession(self, *args, **kwargs):
+        yield self.createSession(*args, **kwargs)
```

Comment:
So this defaults to one global session by default right? No matter how many client objects I create they all share the same session?
I suspect so, I just want to be sure :)

Maybe a nice to have would be a constructor for `AsyncClient` that accepts a session... But maybe that's a thing we add if it's ever needed..